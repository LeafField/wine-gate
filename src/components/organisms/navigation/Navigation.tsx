"use client";
import React from "react";
import CategoryPanel from "../../molecules/category-panel/CategoryPanel";
import SearchPanel from "../../molecules/search-panel/SearchForm";
import UserPanel from "../../molecules/user-panel/UserPanel";
import { selectDummyData } from "../../../utils/dummyData";
import { categoryDummyData } from "../../../utils/dummyData";
import NavMenu from "../../molecules/nav-menu/NavMenu";
import { useStore } from "../../../store";

const Navigation = () => {
  const { menu } = useStore();
  return (
    <nav className="nav">
      <div
        className={`${menu === "close" ? "invisible" : "show-panel"} z-nav tablet:visible tablet:static tablet:space-y-6`}
        id="navigation"
      >
        <UserPanel />
        <CategoryPanel categories={categoryDummyData} />
        <SearchPanel selectData={selectDummyData} />
      </div>
      <NavMenu />
    </nav>
  );
};

export default Navigation;
