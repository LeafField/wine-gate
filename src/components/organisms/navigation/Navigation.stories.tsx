import { Meta, StoryObj } from "@storybook/react";
import Navigation from "./Navigation";
import { spyOn, within } from "@storybook/test";
import { supabase } from "../../../utils/supabase";

const meta = {
  title: "organisms/Navigation",
  component: Navigation,
  parameters: {
    nextjs: {
      appDirectory: true,
    },
  },
} as Meta<typeof Navigation>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  play: async ({ canvasElement }) => {
    within(canvasElement);
    spyOn(supabase.auth, "signOut").mockImplementationOnce(() => {
      console.log("signOut");
      return Promise.resolve({ error: null, data: null });
    });
  },
};
