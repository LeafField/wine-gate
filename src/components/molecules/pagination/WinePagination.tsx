"use client";
import React, { FC } from "react";
import { useRouter } from "next/navigation";
import { Pagination } from "@mantine/core";

type Props = {
  activePage?: number;
  totalPages: number;
  categorySlug: string;
  sort: string;
};

const WinePagination: FC<Props> = ({
  activePage = 1,
  totalPages,
  categorySlug,
  sort,
}) => {
  const router = useRouter();

  const changeHandler = (value: number) => {
    router.push(`/category/${categorySlug}/${sort}/${value}`);
  };
  return (
    <div className="flex justify-center">
      <Pagination
        total={totalPages}
        value={activePage}
        onChange={changeHandler}
      />
    </div>
  );
};

export default WinePagination;
