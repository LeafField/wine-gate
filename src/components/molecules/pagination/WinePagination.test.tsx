import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import WinePagination from "./WinePagination";
import { MantineProvider } from "@mantine/core";
import * as router from "next/navigation";

jest.mock("next/navigation");

describe("WinePagination", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  afterAll(() => {
    jest.restoreAllMocks();
  });

  test("正しくレンダリングがされている", () => {
    const { container } = render(
      <WinePagination totalPages={10} categorySlug="red" sort="new" />,
      {
        wrapper: MantineProvider,
      },
    );
    expect(screen.getByText("1")).toBeInTheDocument();
    expect(screen.getByText("2")).toBeInTheDocument();
    expect(screen.getByText("3")).toBeInTheDocument();
    expect(screen.getByText("4")).toBeInTheDocument();
    expect(screen.getByText("5")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("ページネーションがクリックされた時にrouter.pushが呼ばれる", async () => {
    expect.assertions(1);
    const mockPush = jest.fn();
    jest.spyOn(router, "useRouter").mockImplementationOnce(() => ({
      push: mockPush,
      back: jest.fn(),
      prefetch: jest.fn(),
      replace: jest.fn(),
      forward: jest.fn(),
      refresh: jest.fn(),
    }));
    render(<WinePagination totalPages={10} categorySlug="red" sort="new" />, {
      wrapper: MantineProvider,
    });
    await userEvent.click(screen.getByText("2"));
    return expect(mockPush).toHaveBeenCalledWith("/category/red/new/2");
  });
});
