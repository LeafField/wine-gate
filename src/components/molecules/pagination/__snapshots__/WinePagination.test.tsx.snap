// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WinePagination 正しくレンダリングがされている 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="flex justify-center"
  >
    <div
      class="m-4addd315 mantine-Pagination-root"
    >
      <div
        class="m-4081bf90 mantine-Group-root"
        style="--group-gap: calc(0.5rem * var(--mantine-scale)); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
      >
        <button
          class="mantine-focus-auto m-326d024a mantine-Pagination-control m-87cf2631 mantine-UnstyledButton-root"
          data-disabled="true"
          disabled=""
          type="button"
        >
          <svg
            viewBox="0 0 16 16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.219 8l3.3 3.3-.943.943L5.333 8l4.243-4.243.943.943-3.3 3.3z"
              fill="currentColor"
            />
          </svg>
        </button>
        <button
          aria-current="page"
          class="mantine-focus-auto mantine-active m-326d024a mantine-Pagination-control m-87cf2631 mantine-UnstyledButton-root"
          data-active="true"
          data-with-padding="true"
          type="button"
        >
          1
        </button>
        <button
          class="mantine-focus-auto mantine-active m-326d024a mantine-Pagination-control m-87cf2631 mantine-UnstyledButton-root"
          data-with-padding="true"
          type="button"
        >
          2
        </button>
        <button
          class="mantine-focus-auto mantine-active m-326d024a mantine-Pagination-control m-87cf2631 mantine-UnstyledButton-root"
          data-with-padding="true"
          type="button"
        >
          3
        </button>
        <button
          class="mantine-focus-auto mantine-active m-326d024a mantine-Pagination-control m-87cf2631 mantine-UnstyledButton-root"
          data-with-padding="true"
          type="button"
        >
          4
        </button>
        <button
          class="mantine-focus-auto mantine-active m-326d024a mantine-Pagination-control m-87cf2631 mantine-UnstyledButton-root"
          data-with-padding="true"
          type="button"
        >
          5
        </button>
        <div
          class="m-4ad7767d mantine-Pagination-dots"
        >
          <svg
            viewBox="0 0 16 16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2 8c0-.733.6-1.333 1.333-1.333.734 0 1.334.6 1.334 1.333s-.6 1.333-1.334 1.333C2.6 9.333 2 8.733 2 8zm9.333 0c0-.733.6-1.333 1.334-1.333C13.4 6.667 14 7.267 14 8s-.6 1.333-1.333 1.333c-.734 0-1.334-.6-1.334-1.333zM6.667 8c0-.733.6-1.333 1.333-1.333s1.333.6 1.333 1.333S8.733 9.333 8 9.333 6.667 8.733 6.667 8z"
              fill="currentColor"
            />
          </svg>
        </div>
        <button
          class="mantine-focus-auto mantine-active m-326d024a mantine-Pagination-control m-87cf2631 mantine-UnstyledButton-root"
          data-with-padding="true"
          type="button"
        >
          10
        </button>
        <button
          class="mantine-focus-auto mantine-active m-326d024a mantine-Pagination-control m-87cf2631 mantine-UnstyledButton-root"
          type="button"
        >
          <svg
            viewBox="0 0 16 16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.781 8l-3.3-3.3.943-.943L10.667 8l-4.243 4.243-.943-.943 3.3-3.3z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
